test/unit/error.spec.ts

```diff
@@ -292,3 +285,88 @@ test("return from nested finally", () => {
     `;
     expect(util.transpileAndExecute(code)).toBe("finally AB");
 });
+
+test("throw and catch custom error object", () => {
+    util.testFunction`
+        try {
+            throw { x: "Hello error object!" };
+        } catch (error) {
+            return error.x;
```

Comment:
```suggestion
            return error;
```

It would be easier to test other types with `.each` and also would check that we don't change thrown value in any way